assets/js/components/relationship-manager.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { ContentPicker } from '@10up/block-components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { store } from '../store';
+import { ContentConnectRelationship } from '../store/types';
+
+type RelationshipManagerProps = {
+	postId: number | null;
+	relationship: ContentConnectRelationship;
+};
+
+export function RelationshipManager({ postId, relationship }: RelationshipManagerProps) {
+	const { updateRelatedEntities } = useDispatch(store);
+
+	const { relatedEntities } = useSelect((select) => ({
+		relatedEntities: select(store).getRelatedEntities(postId, {
+			rel_key: relationship.rel_key,
+			rel_type: relationship.rel_type,
+		}),
+	}), [postId, relationship.rel_key]);
+
+	const handleChange = async (newEntities: any[]) => {
+		await updateRelatedEntities(
+			postId,
+			relationship.rel_key,
+			relationship.rel_type,
+			newEntities
+		);
+	};
+
+	return (
+		<ContentPicker
+			onPickChange={handleChange}
+			mode={relationship?.object_type ?? 'post'}
+			content={relatedEntities}
+			contentTypes={relationship?.post_type}
+			maxContentItems={relationship?.max_items ?? 100}
+			isOrderable={relationship?.sortable ?? false}
+			queryFilter={(query) => {
+				if (relationship?.rel_key) {
+					return addQueryArgs(query, {
+						content_connect: relationship.rel_key
```

Comment:
I added this search parameter mainly to give context to the editor-side search when it reaches the `rest_post_search_query` hook in PHP. One of my clients uses ElasticPress, and I needed a way to disable EP for these specific queries. This parameter can also be used to filter search results based on client-specific requirements. It might also be worth including the current post ID.